geopyspark/geotrellis/key_conversion.py

```diff
@@ -0,0 +1,119 @@
+        self.__jvm = gps.get_spark_context()._gateway.jvm
+
+        if isinstance(layout, gps.LocalLayout):
+            if not extent:
+                raise ValueError("Must specify an extent when using LocalLayout")
+
+            if dimensions and not cellsize:
+                cellsize = ((extent.xmax - extent.xmin)/dimensions[0], (extent.ymax - extent.ymin)/dimensions[1])
+                dimensions = None
+
+            if cellsize and not dimensions:
+                tilewidth = layout.tile_cols * cellsize[0]
+                tileheight = layout.tile_rows * cellsize[1]
+                rows = ceil((extent.xmax - extent.xmin) / tilewidth)
+                cols = ceil((extent.ymax - extent.ymin) / tileheight)
+                extent = gps.Extent(extent.xmin, extent.ymax - rows * tileheight, extent.xmin + cols * tilewidth, extent.ymax)
+                tl = gps.TileLayout(cols, rows, layout.tile_cols, layout.tile_rows)
+            else:
+                raise ValueError("For LocalLayout, must specify exactly one: cellsize or dimension")
+        elif isinstance(layout, gps.GlobalLayout):
+            from pyproj import Proj, transform
+
+            if not crs:
+                raise ValueError("Must specify a crs when using GlobalLayout")
+
+            if isinstance(crs, int):
+                crs = "{}".format(crs)
+
+            gtcrs = self.__jvm.geopyspark.geotrellis.TileLayer.getCRS(crs).get()
+
+            if gtcrs.epsgCode().isDefined() and gtcrs.epsgCode().get() == 3857:
+                extent = gps.Extent(-20037508.342789244, -20037508.342789244, 20037508.342789244, 20037508.342789244)
+            elif gtcrs.epsgCode().isDefined() and gtcrs.epsgCode().get() == 4326:
+                extent = gps.Extent(-180.0, -89.99999, 179.99999, 89.99999)
+            else:
+                llex = gps.Extent(-180.0, -89.99999, 179.99999, 89.99999)
+                proj4str = gtcrs.toProj4String()
+                target = Proj(proj4str)
+                xmin, ymin = target(llex.xmin, llex.ymin)
+                xmax, ymax = target(llex.xmax, llex.ymax)
+                extent = gps.Extent(xmin, ymin, xmax, ymax)
+
+            if not layout.zoom:
+                raise ValueError("Must specify a zoom level")
+
+            layout_rows_cols = int(pow(2, layout.zoom))
+            tl = gps.TileLayout(layout_rows_cols, layout_rows_cols, layout.tile_size, layout.tile_size)
+        elif isinstance(layout, gps.LayoutDefinition):
+            extent = layout.extent
+            tl = layout.tileLayout
+
+        ex = self.__jvm.geotrellis.vector.Extent(float(extent.xmin), float(extent.ymin), float(extent.xmax), float(extent.ymax))
+        tilelayout = self.__jvm.geotrellis.raster.TileLayout(int(tl[0]), int(tl[1]), int(tl[2]), int(tl[3]))
+        self.__layout = self.__jvm.geotrellis.spark.tiling.LayoutDefinition(ex, tilelayout)
+
+    def keyToExtent(self, key, *args):
```

Comment:
This needs a docstring

Reply:
Also, Python methods use snake case, so the method should be called, `key_to_extent`